### PR TITLE
fix(config): Migrate semantic commit option to enum type

### LINKED ISSUE #7163: refactor semanticCommits from boolean to enum
Change from true/false/null to 'enabled'/'disabled'/'auto'. Keep default as 'auto' for now. Needs config migrating too plus need to think what to do about inherited presets which use `boolean`.

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7163

### CODE REVIEW

File: lib/config/common.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export interface RenovateSharedConfig {
   repositoryCache?: RepositoryCacheConfig;
   requiredStatusChecks?: string[];
   schedule?: string[];
-  semanticCommits?: boolean;
+  semanticCommits?: 'auto' | 'enabled' | 'disabled';
```
Comment (rarkins) [inline lib/config/common.ts:55]: Are we using this anywhere else? Perhaps make it a type?

**Review by rarkins — COMMENTED** [review]
Another possibility: convert to "semanticCommitsEnabled" true/false internally once we resolve the value of semanticCommits? Not sure if it's better or not

File: lib/config/common.ts (hunk 1 above)
Comment (zharinov) [reply lib/config/common.ts:55]: Do you mean the triple pattern of `auto/enabled/disabled`?

Comment (rarkins) [reply lib/config/common.ts:55]: OK

**Comment by zharinov** [comment]
Okay, I can confirm it migrated preset [from this fixture](https://raw.githubusercontent.com/renovatebot/renovate/master/lib/config/presets/__fixtures__/renovate-config-ikatyang.json) which I left untouched.

**Comment by Turbo87** [comment]
just out of curiosity, why was this labels as `fix` instead of `feat`?

**Comment by zharinov** [comment]
Can't remember, most probably just confusion 🙃

**Pull request merged**